lib/vmdb/plugins.rb

```diff
@@ -47,6 +47,17 @@ def versions
       details.transform_values { |v| v[:version] }
     end
 
+    def plugin_for_class(klass)
+      klass = klass.to_s unless klass.kind_of?(String)
+
+      klass_path, _klass_line = Object.const_source_location(klass)
+      return if klass_path.nil?
+
+      paths.each do |engine, path|
+        return engine if klass_path.start_with?(path)
+      end
```

Comment:
Think this is cleaner than .each:

```suggestion
      found, = paths.detect { |_engine, path|  klass_path.start_with?(path) }
      found
```